spec/services/billing_service_spec.rb

```diff
@@ -6,19 +6,25 @@
   subject(:billing_service) { described_class.new }
 
   describe '.call' do
-    let(:start_date) { DateTime.parse('20 Feb 2021') }
+    let(:plan) { create(:plan, interval: interval, bill_charges_monthly: bill_charges_monthly) }
+    let(:bill_charges_monthly) { false }
+    let(:subscription_date) { DateTime.parse('20 Feb 2021') }
+
+    let(:subscription) do
+      create(
+        :subscription,
+        plan: plan,
+        subscription_date: subscription_date,
+        started_at: Time.zone.now,
+        billing_time: billing_time,
+      )
+    end
 
-    context 'when billed weekly' do
-      let(:plan) { create(:plan, interval: :weekly) }
+    before { subscription }
```

Comment:
why not using `let!`?

Reply:
Because Rubocop :) 
Actually I think it makes sense because it reduces the "magic effect"